feat(FR-48): WebUI NEO start page basic layout

ironAiken2 opened the pull request:
<!--
Please precisely, concisely, and concretely describe what this PR changes, the rationale behind codes,
and how it affects the users and other developers.
-->

### This PR is about UI changing for `Backend.AI WebUI NEO` #2835 
(FR-48)

**Changes:**
- Added a start page utilizing the BAIBoard component.
- Added `ActionItemContent` component (can be used as primary action card for start page and subsequent pages)
- For responsive components, delete the descriptions of all cards if even one card causes scrolling. 

> ! note
> - The Cloudscape board component uses the Grid component internally. The braek point of a grid component is calculated based on the size of the container, not the window size, as follows [ref](https://github.com/cloudscape-design/components/blob/a528836d4537ec132b6c4af4f920db37fc22219e/src/internal/breakpoints.ts#L5-L13)
> - The Cloudscape board component used for BAIBaord does not yet provide the ability to adjust columns per screen break point. [ref](https://github.com/cloudscape-design/components/discussions/1339)

**How to test:**
- Navigate to start page (/start) 
- Verify that the Button for each item is working properly. 
- Verify that each item is repositioned correctly. (The Resize option is currently not allowed.
- In this PR, the default page has been changed from summary to start. We need to check if the redirect from error page to start is working properly, or if the refresh is working properly, etc.

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/lSyr8xXz1wdXALkJKzVx/6990669d-423f-4ed8-a96e-47ad163bdb16.png)

**Checklist:** (if applicable)

- [ ] Mention to the original issue
- [ ] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

ironAiken2 commented:
* **#2786** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2786?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ironAiken2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

agatha197 commented on react/src/pages/StartPage.tsx:
```diff
@@ -0,0 +1,227 @@
+      // content: <SummaryItemStartMenu allowNeoSessionLauncher />,
+      content: (
+        <ActionItemContent
+          title="Create a New Storage Folder"
+          description="Creating a folder and uploading files are crucial for training models or providing services externally. Please begin by uploading the files."
+          buttonText="Create Folder"
```
Let's use i18n

ironAiken2 replied:
Added translation. The wording is not finalized yet, so we need @YEONFEEL96  to confirm.

Can you please check the wording of en.json and en.json under resource/i18n? @YEONFEEL96

agatha197 reviewed (changes requested):
The interactive type is selected when I click the `Start Batch Session` button.

ironAiken2 commented:
> The interactive type is selected when I click the `Start Batch Session` button.

resolved!

YEONFEEL96 commented on resources/i18n/en.json:
```diff
@@ -1757,5 +1758,26 @@
+      "StartBatchSession": "Start from a URL"
+    },
+    "CreateFolderDesc": "Creating a folder and uploading files are crucial for training models or providing services externally. Please begin by uploading the files.",
+    "StartSession": "Start an Interactive Session",
```
```suggestion
    "StartSession": "Start Interactive Session",
```

YEONFEEL96 commented on resources/i18n/en.json:
```diff
@@ -1757,5 +1758,26 @@
     "CompareWithOtherModels": "Compare with other models"
+  },
+  "start": {
+    "CreateFolder": "Create a New Storage Folder",
```
```suggestion
    "CreateFolder": "Create New Storage Folder",
```

YEONFEEL96 commented on resources/i18n/en.json:
```diff
@@ -1757,5 +1758,26 @@
+      "StartNow": "Start Now",
+      "StartBatchSession": "Start from a URL"
+    },
+    "CreateFolderDesc": "Creating a folder and uploading files are crucial for training models or providing services externally. Please begin by uploading the files.",
```
```suggestion
    "CreateFolderDesc": "Create folders and upload files. To train models or provide external services, it is essential to create folders and upload files.",
```

YEONFEEL96 commented on resources/i18n/en.json:
```diff
@@ -1757,5 +1758,26 @@
+    },
+    "CreateFolderDesc": "Creating a folder and uploading files are crucial for training models or providing services externally. Please begin by uploading the files.",
+    "StartSession": "Start an Interactive Session",
+    "StartSessionDesc": "Want to train a model? Click the button to create a session instantly. Choose your preferred environment and resources for running code and other tasks.",
```
```suggestion
    "StartSessionDesc": "Create a session to train a model. Choose your preferred environment and resources to run a code.",
```

YEONFEEL96 commented on resources/i18n/en.json:
```diff
@@ -1757,5 +1758,26 @@
+    "StartSession": "Start an Interactive Session",
+    "StartSessionDesc": "Want to train a model? Click the button to create a session instantly. Choose your preferred environment and resources for running code and other tasks.",
+    "StartBatchSession": "Start Batch Session",
+    "StartBatchSessionDesc": "For predefined files or scheduled tasks, try Batch sessions. Enter the command, set the date and time, and run as needed.",
```
```suggestion
    "StartBatchSessionDesc": "Create a batch session for predefined files or scheduled tasks. Enter the command, set the date and time, and run the session on demand.",
```

YEONFEEL96 commented on resources/i18n/en.json:
```diff
@@ -1757,5 +1758,26 @@
+    "StartBatchSession": "Start Batch Session",
+    "StartBatchSessionDesc": "For predefined files or scheduled tasks, try Batch sessions. Enter the command, set the date and time, and run as needed.",
+    "ModelService": "Start Model Service",
+    "ModelServiceDesc": "Ready to share your trained model with others? Click the button to create a service.",
```
```suggestion
    "ModelServiceDesc": "Select to share your trained model with others. This action will create a model service.",
```

YEONFEEL96 commented on resources/i18n/ko.json:
```diff
@@ -1754,5 +1755,26 @@
+  "start": {
+    "CreateFolder": "새 스토리지 폴더 생성",
+    "button": {
+      "CreateFolder": "폴더 생성",
```
```suggestion
      "CreateFolder": "폴더 생성하기",
```

YEONFEEL96 commented on resources/i18n/ko.json:
```diff
@@ -1754,5 +1755,26 @@
     "CompareWithOtherModels": "다른 모델과 비교하기"
+  },
+  "start": {
+    "CreateFolder": "새 스토리지 폴더 생성",
```
```suggestion
    "CreateFolder": "새 스토리지 폴더 생성하기",
```

YEONFEEL96 commented on resources/i18n/ko.json:
```diff
@@ -1754,5 +1755,26 @@
+      "StartSession": "세션 생성",
+      "ModelService": "서비스 생성",
+      "StartNow": "지금 시작하기",
+      "StartBatchSession": "URL 로 시작"
```
```suggestion
      "StartBatchSession": "URL로 시작하기"
```

YEONFEEL96 commented on resources/i18n/ko.json:
```diff
@@ -1754,5 +1755,26 @@
+      "StartNow": "지금 시작하기",
+      "StartBatchSession": "URL 로 시작"
+    },
+    "CreateFolderDesc": "폴더를 생성하고 파일을 업로드 하는 것은 모델 학습 또는 서비스 제공에 중요한 역할을 담당합니다. 먼저 파일을 업로드 하세요.",
```
```suggestion
    "CreateFolderDesc": "폴더를 만들고 파일을 업로드합니다. 모델을 학습시키거나 외부 서비스를 제공하려면 해당 작업은 필수적입니다.",
```

YEONFEEL96 commented on resources/i18n/ko.json:
```diff
@@ -1754,5 +1755,26 @@
+    },
+    "CreateFolderDesc": "폴더를 생성하고 파일을 업로드 하는 것은 모델 학습 또는 서비스 제공에 중요한 역할을 담당합니다. 먼저 파일을 업로드 하세요.",
+    "StartSession": "Interactive 세션 생성",
+    "StartSessionDesc": "모델을 학습시키고 싶으신가요? 버튼을 클릭하면 즉시 세션을 생성할 수 있습니다. 코드 실행 및 기타 작업을 위해 원하는 환경과 자우너을 선택하세요. ",
```
```suggestion
    "StartSessionDesc": "모델을 학습시킬 세션을 생성합니다. 원하는 환경과 리소스를 선택하여 코드를 실행하세요.",
```

YEONFEEL96 commented on resources/i18n/ko.json:
```diff
@@ -1754,5 +1755,26 @@
+    "StartSession": "Interactive 세션 생성",
+    "StartSessionDesc": "모델을 학습시키고 싶으신가요? 버튼을 클릭하면 즉시 세션을 생성할 수 있습니다. 코드 실행 및 기타 작업을 위해 원하는 환경과 자우너을 선택하세요. ",
+    "StartBatchSession": "Batch 세션 생성",
+    "StartBatchSessionDesc": "미리 정의된 파일 또는 예약된 작업의 경우, 배치 세션을 사용해 보세요. 명령을 입력하고 날짜와 시간을 설정한 다음 필요에 따라 실행합니다. ",
```
```suggestion
    "StartBatchSessionDesc": "미리 정의된 파일 또는 예약된 작업에 대한 배치 세션을 생성합니다. 명령을 입력하고 날짜와 시간을 설정한 다음 필요에 따라 세션을 실행하세요.",
```

YEONFEEL96 commented on resources/i18n/ko.json:
```diff
@@ -1754,5 +1755,26 @@
+    "StartBatchSession": "Batch 세션 생성",
+    "StartBatchSessionDesc": "미리 정의된 파일 또는 예약된 작업의 경우, 배치 세션을 사용해 보세요. 명령을 입력하고 날짜와 시간을 설정한 다음 필요에 따라 실행합니다. ",
+    "ModelService": "모델 서비스 시작",
+    "ModelServiceDesc": "학습된 모델을 다른 사람들과 공류할 준비가 되었나요? 버튼을 클릭하여 서비스를 생성하세요.",
```
```suggestion
    "ModelServiceDesc": "학습이 끝난 모델을 다른 사람과 공유하고 싶다면, 버튼을 클릭하여 서비스를 생성하세요.",
```

YEONFEEL96 commented on resources/i18n/ko.json:
```diff
@@ -1754,5 +1755,26 @@
+    "ModelService": "모델 서비스 시작",
+    "ModelServiceDesc": "학습된 모델을 다른 사람들과 공류할 준비가 되었나요? 버튼을 클릭하여 서비스를 생성하세요.",
+    "StartFromExample": "가져오기 및 실행",
+    "StartFromExampleDesc": "다양한 예제 코드를 살펴보고 실행하세요. ",
```
```suggestion
    "StartFromExampleDesc": "다양한 예제 코드를 살펴보고, 해당 코드로부터 실행할 수 있습니다.",
```

YEONFEEL96 commented on resources/i18n/ko.json:
```diff
@@ -1754,5 +1755,26 @@
+    "StartFromExample": "가져오기 및 실행",
+    "StartFromExampleDesc": "다양한 예제 코드를 살펴보고 실행하세요. ",
+    "StartFromURL": "URL 로 시작",
+    "StartFromURLDesc": "GitHub, GitLab, Notebook 등 다양한 환경에서 프로젝트와 코드를 가져오세요. "
```
```suggestion
    "StartFromURLDesc": "GitHub, GitLab, Notebook 등 다양한 환경에서 프로젝트와 코드를 가져오고, 실행할 수 있습니다. "
```

YEONFEEL96 reviewed (changes requested):
Improved English and Korean description of each feature / button which shown in UI.
For ko-KR, we will limit the use of '~하기' in overall cases.

YEONFEEL96 reviewed (approved):
Confirmed that my changes have been reflected. LGTM.

agatha197 commented on react/src/pages/StartPage.tsx:
```diff
@@ -0,0 +1,244 @@
+            const params = new URLSearchParams();
+            params.set('step', '0');
+            params.set('formValues', JSON.stringify(launcherValue));
+            console.log(launcherValue);
+            console.log(params);
```
```suggestion
```

agatha197 commented on react/src/pages/StartPage.tsx:
```diff
@@ -0,0 +1,244 @@
+            <AppstoreAddOutlined
+              style={{
+                fontSize: token.fontSizeHeading3,
+                color: token['cyan-6'],
```
Let's add a comment to change this with token later. And it would better put the same comment on the similar code below to easily change it.

ironAiken2 replied:
It's provided with the default antd token color, but it would be nice to add a specific token as a provider, I'll leave a comment!

yomybaby reviewed (changes requested):
Let's remove the default padding of `BAIBoardItem`. This will make it easier to customize and manage scroll position.

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/XqC2uNFuj0wg8I60sMUh/65f2a5c5-e29f-4592-a724-1331582860ac.png)

ironAiken2 commented:
> Let's remove the default padding of `BAIBoardItem`. This will make it easier to customize and manage scroll position.
> 
> ![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/XqC2uNFuj0wg8I60sMUh/65f2a5c5-e29f-4592-a724-1331582860ac.png)
> 
> 

Currently, we don't show a description when a scroll occurs, so the scroll itself shouldn't occur.

ironAiken2 commented:
I tried to remove the default padding for board-item, but there are some issues.

patch-package offical docs recommend using npm or yarn, and pnpm patch for pnpm. Since patch-package is patching based on package-lock.json, I don't think it's possible to use it with the current structure using pnpm. @yomybaby

yomybaby commented:
> I tried to remove the default padding for board-item, but there are some issues.
> 
> �patch-package offical docs recommend using npm or yarn, and pnpm patch for pnpm. Since patch-package is patching based on package-lock.json, I don't think it's possible to use it with the current structure using pnpm. @yomybaby

There is draft PR which I created, please check it.
https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2661/chore-use-pnpm-patch-in-react-instead-of-patch-package
[Teams thread](https://teams.microsoft.com/l/message/19:14c484402d874dafb15806d093b95a82@thread.skype/1724401071604?tenantId=13c6a44d-9b52-4b9e-aa34-0513ee7131f2&groupId=74ae2c4d-ec4d-4fdf-b2c2-f5041d1e8631&parentMessageId=1724401071604&teamName=devops&channelName=Frontend&createdTime=1724401071604)

ironAiken2 commented:
I tested it on [pnpm patch PR](https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2661/chore-use-pnpm-patch-in-react-instead-of-patch-package) and found it works fine, so I modified it to work based on pnpm patch. I've also fixed the scroll position issue. @yomybaby

yomybaby reviewed (approved):
LGTM

yomybaby reviewed (changes requested):
It would be better
- Prevent moving items for now (let's enable this later).
- Arrange each item by color: orange items in the first row and green items in the second row.
- Delete the Start menu from the Summary page.

yomybaby reviewed (approved):
LGTM
